### PR TITLE
.github: new Github workflow for open-vm-tools

### PR DESCRIPTION
**Pull request by dongsupark**
Automatically update `coreos/open-vm-tools` as well as `coreos-base/oem-vmware`.
Get the latest open-vm-tools release number, and get its build number from the Github repo, and replace the old build number with the new one.
Also sync `coreos-base/oem-vmware` in line with `open-vm-tools`.

To do that, we need to also split git repo init part into `prepare_git_repo`, to prevent some corner cases, where applying multiple patches does not work because the latter overwrites the former patch.
We need to also clean up an unnecessary patch file.

### CODE REVIEW

File: .github/workflows/vmware-releases-main.yml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+          PATH: ${{ steps.setup-flatcar-sdk.outputs.path }}
+          VERSION_NEW: ${{ steps.fetch-latest-release.outputs.VERSION_MAIN }}
+        run: .github/workflows/vmware-apply-patch.sh
+      - name: Create pull request for main
+        uses: peter-evans/create-pull-request@v3
+        if: steps.apply-patch-main.outputs.UPDATE_NEEDED == 1
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+          base: ${{ steps.fetch-latest-release.outputs.BASE_BRANCH_MAIN }}
+          branch: vmware-${{ steps.fetch-latest-release.outputs.VERSION_MAIN }}-main
```
Comment (jepio) [inline .github/workflows/vmware-releases-main.yml:41]: This `main` should be `${{ steps.fetch-latest-release.outputs.BASE_BRANCH_MAIN }}`

File: .github/workflows/vmware-releases-main.yml [hunk 2]
```diff
@@ -0,0 +1,47 @@
+name: Get the latest open-vm-tools release for main
+on:
+  schedule:
+    - cron:  '0 7 * * 5'
+
```
Comment (jepio) [inline .github/workflows/vmware-releases-main.yml:5]: If you add ` workflow_dispatch:` as a trigger, then we can manually test it right after merging.

**Review by jepio — APPROVED** [review]
Looks good, had just two minor comment, but i think it can be merged

File: .github/workflows/vmware-releases-main.yml (hunk 1 above)
Comment (dongsupark) [reply .github/workflows/vmware-releases-main.yml]: Right, done

File: .github/workflows/vmware-releases-main.yml (hunk 2 above)
Comment (dongsupark) [reply .github/workflows/vmware-releases-main.yml:6]: Good idea, added workflow_dispatch to every workflow.

**Pull request merged**